tests/TagTest.php

```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
         ]);
         $this->assertSame('<img src="IMAGE.PNG" id="123" data-chars="&quot;\'&lt;&gt;">', $tag->render());
     }
+
+    public function testHasAttribute()
+    {
+        $tag = new Tag('p');
+        $tag->setAttributes([
+            'class' => 'class_name'
+        ]);
+
+        $this->assertTrue($tag->hasAttribute('class'));
+        $this->assertFalse($tag->hasAttribute('invalid_attribute'));
+    }
+
+    public function testGetAttribute()
+    {
+        $tag = new Tag('p');
+        $tag->setAttributes([
+            'class' => 'class_name'
+        ]);
+
+        $this->assertSame('class_name', $tag->getAttribute('class'));
+        $this->assertNull($tag->getAttribute('invalid_attribute'));
+    }
+
+    public function testRemoveAttribute()
+    {
+        $tag = new Tag('p');
+        $tag->setAttributes([
+            'class' => 'class_name'
+        ]);
+
+        $this->assertInstanceOf(Tag::class, $tag->removeAttribute('class'));
```

Comment:
You could actually use `assertSame($tag, ...` to check that the same instance is returned!